tests/test_memory_repr.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+import pytest
+import torch
+import cupy as cp
+from savant.utils.memory_repr import opencv_gpu_mat_as_cupy, cupy_as_opencv_gpu_mat
+from savant.utils.memory_repr_pytorch import opencv_gpu_mat_as_pytorch, pytorch_tensor_as_opencv_gpu_mat
+import cv2
+
+TORCH_TYPE = [torch.int8, torch.uint8, torch.float32]
+NUMPY_TYPE = [np.int8, np.uint8, np.float32]
+CUPY_TYPE = [cp.int8, cp.uint8, cp.float32]
+
+
+def get_opencv_image(input_type, channels=3):
+
+    if channels == 3:
+        image_np = np.random.randint(0, 255, (10, 20, 3)).astype(input_type)
+    elif channels == 1:
+        image_np = np.random.randint(0, 255, (10, 20)).astype(input_type)
```

Comment:
Is it possible that we will get a shape (H, W, 1)? Should we process this case?

(related to any input data, opencv, torch, cupy)

Reply:
It's possible. The cases have been added to the tests